compiler/rustc_ast/src/token.rs

```diff
@@ -80,6 +84,38 @@ pub struct Lit {
     pub suffix: Option<Symbol>,
 }
 
+impl Lit {
+    /// Returns `true` if this is a float literal. This includes ones like
+    /// `1f32` that have an `Integer` kind but are truly floats.
```

Comment:
There's no "truly" here, lexically they are integer tokens and not floats at all.